alacritty/src/window.rs

```diff
@@ -361,6 +368,26 @@ impl Window {
         self.window().wayland_display()
     }
 
+    #[cfg(not(any(target_os = "macos", target_os = "windows")))]
+    pub fn set_wayland_theme(&mut self, colors: &TerminalColors) {
+        let hovered_close_icon = colors.normal().red;
+        let hovered_maximize_icon = colors.normal().green;
+        let hovered_minimize_icon = colors.normal().yellow;
+        let foreground = colors.primary.foreground;
+        let background = colors.primary.background;
+        let dim_foreground = colors.primary.dim_foreground.unwrap_or(foreground * DIM_FACTOR);
+
+        let theme = AlacrittyWaylandTheme {
+            foreground,
+            background,
+            dim_foreground,
+            hovered_close_icon,
+            hovered_minimize_icon,
+            hovered_maximize_icon,
+        };
```

Comment:
There's a lot of crap in this function body just taking things from `colors` and putting it inside of `AlacrittyWaylandTheme`. Could we just make a `new` function for the theme which takes a color, so that `window.rs` is less impacted?